src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests.cs

```diff
@@ -3526,6 +3526,28 @@ public static void Main()
         }
 
         [Fact]
+        public void ThrowExpressionWithNullableDecimal()
+        {
+            var source = @"
+using System;
+public class ITest
+{
+    public decimal Test() => 1m;
+}
+
+public class TestClass
+{
+    public void Test(ITest test)
+    {
+        var result = test?.Test() ?? throw new Exception();
+    }
+}";
+            var compilation = CreateCompilationWithMscorlib(source, options: TestOptions.DebugDll);
+            compilation.VerifyDiagnostics();
+            compilation.VerifyEmitDiagnostics();
```

Comment:
>compilation.VerifyEmitDiagnostics(); [](start = 12, length = 36)

Please adjust the test scenario so that it can be executed and call ```public void Test(ITest test)``` with null and not null input, observe expected behavior of the program.